packages/host/app/commands/send-ai-assistant-message.ts

```diff
@@ -106,10 +106,19 @@ export default class SendAiAssistantMessageCommand extends HostBaseCommand<
     if (files?.length) {
       files = await this.matrixService.uploadFiles(files);
     }
-    let cardFileDefs = await matrixService.uploadCardsAndUpdateSkillCommands(
-      input.attachedCards ?? [],
-      roomId,
+    // Upload skill cards and attached cards together to ensure:
+    // 1. Latest skill cards are deployed with the message
```

Comment:
It is awkward that we are relying on the caller of `uploadCardsAndUpdateSkillCommands` to include the room's skill cards? Why not make that an internal implementation detail of the method?

Reply:
I removed the `uploadCardsAndUpdateSkillCommands` and created a new function in matrix-service called `updateSkillsAndCommandsIfNeeded`. Please check it again.